### ISSUE TITLE
Unexpected behavior with links containing query params

### ISSUE THREAD

**Issue by oski646** [inertiajs/inertia#1497]
### Version:

- `@inertiajs/vue2` version: 1.0.2

### Describe the problem:

There is always `&` added at the end of the URL. This change actually broke some of the logic in the project (active classes on links). The expected behavior is the URL without `&` at the end. I was on version 0.10.1 before upgrading to 1.0.2 and it worked well.

### Steps to reproduce:

Just add the `Link` component into the `template` with `href="/dashboard?some=query"`. It will redirect you to the `/dashboard?some=query&`.

**Comment by RobertBoes** [comment]
Might be worth sharing a repo that reproduces the issue or a code-snippet of what you're doing exactly. Using Vue 3 and I can't replicate the issue. The Link component for Vue 2 and Vue 3 are pretty much equal, and both use this snippet to get the correct href:
```js
const [href, propsData] = mergeDataIntoQueryString(
    method,
    props.href || '',
    props.data,
    props.queryStringArrayFormat,
)
```
(https://github.com/inertiajs/inertia/blob/master/packages/vue2/src/link.ts#L93-L98 and https://github.com/inertiajs/inertia/blob/master/packages/vue3/src/link.ts#L75)

Using this navigates correctly:
```html
<template>
  <Link href="/dashboard?some=query">Click me!</Link>
</template>
```

**Comment by oski646** [comment]
Yeah, I think I found the issue. I'm currently upgrading some dependencies in the project and one of them is `axios`. I had problems with `axios` and I had to change the default params serializer `axios.defaults.paramsSerializer`. After commenting out my changes to the custom paramsSerializer the problem is gone.

So, everything with Inertia.js seems to be working as expected but somehow `axios` default configuration influences Inertia.js navigation.

**Comment by RobertBoes** [comment]
Yeah I think that would be expected, as Inertia uses Axios to make requests, so if you change the default configuration that would have an impact on the calls that Inertia makes

**Comment by oski646** [comment]
Ok, so all is fine. Now I have to deal with `axios`. Thanks that you tested it on your end, so I knew that I have to find the issue somewhere else.
